djangocms_navigation/admin.py

```diff
@@ -476,33 +469,33 @@ def add_view(self, request, menu_content_id=None, form_url="", extra_context=Non
 
         return super().add_view(request, form_url=form_url, extra_context=extra_context)
 
-    def preview_view(self, request, menu_content_id=None, form_url="", extra_context=None):
+    def preview_view(self, request, menu_content_id, extra_context=None):
+        """
+        Renders a preview of the MenuContent and MenuItem objects in 'readonly' without actions.
+        """
+        self.change_list_template = self.get_changelist_template(request=request)
         extra_context = extra_context or {}
-        if menu_content_id:
-            request.menu_content_id = menu_content_id
-            if self._versioning_enabled:
-                menu_content = get_object_or_404(
-                    self.menu_content_model._base_manager, id=menu_content_id
-                )
-                version = Version.objects.get_for_content(menu_content)
-                try:
-                    version.check_view(request.user)
-                except ConditionFailed as error:
-                    messages.error(request, str(error))
-                    return HttpResponseRedirect(version_list_url(menu_content))
-                # purge menu cache
-                purge_menu_cache(site_id=menu_content.menu.site_id)
 
-            extra_context["list_url"] = reverse(
-                "admin:{}_menuitem_list".format(self.model._meta.app_label),
-                kwargs={"menu_content_id": menu_content_id},
-            )
-        if request.GET.get("content_type_id"):
-            return True
-        else:
-            return False
+        request.menu_content_id = menu_content_id
+        menu_content = get_object_or_404(
+            self.menu_content_model._base_manager, id=menu_content_id
+        )
+        extra_context["title"] = "Preview Menu: {}".format(menu_content.__str__())
```

Comment:
How come we are using .format here? This admin file already uses f strings, so I'd recommend using those.

Reply:
It was because it was lifted from the [changelist_view](https://github.com/django-cms/django-cms/commit/ff6cb9b5dced92eadef62694e989d601e9475b30) which uses `.format`, I hadn't noticed the file using f-strings but I personally prefer it for most circumstances so happy to change.